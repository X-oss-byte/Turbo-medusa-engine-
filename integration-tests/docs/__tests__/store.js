const fixtureWriter = require("../utils/write-fixture").default
const { useApi } = require("../../helpers/use-api")
const { useDb } = require("../../helpers/use-db")

const cartTest = require("../test-input/store/cart")
const customerTest = require("../test-input/store/customer")
const giftCardTest = require("../test-input/store/gift-card")
const productTest = require("../test-input/store/product")
const swapTest = require("../test-input/store/swap")
const regionTest= require("../test-input/store/region")
const orderTest = require('../test-input/store/order')
const toTest = [cartTest, customerTest, giftCardTest, productTest, swapTest, regionTest, orderTest]

jest.setTimeout(30000)

test.each(toTest)(
  "$operationId",
  async ({ setup, buildEndpoint, snapshotMatch, operationId }) => {
    const api = useApi()
    const db = useDb()
    
    const manager = db.connection.manager
    const idOrConfig = await setup(manager, api)
    let config = idOrConfig
    if (typeof idOrConfig === "string") {
      config = {
        id: idOrConfig,
      }
    }

    const response = await api
      .get(buildEndpoint(config.id), {
        headers: config.headers,
      })
      .catch((err) => {
        console.log(err)
      })

    expect(response.data).toMatchSnapshot(snapshotMatch)

    fixtureWriter.addFixture(`store.${operationId}`, response.data)
  }
)
