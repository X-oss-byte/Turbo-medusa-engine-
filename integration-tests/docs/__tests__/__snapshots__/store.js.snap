// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetCartsCart 1`] = `
Object {
  "cart": Object {
    "billing_address": null,
    "billing_address_id": null,
    "completed_at": null,
    "context": Object {
      "ip": "::ffff:127.0.0.1",
      "user_agent": "axios/0.21.1",
    },
    "created_at": Any<String>,
    "customer_id": null,
    "deleted_at": null,
    "discount_total": 0,
    "discounts": Array [],
    "email": null,
    "gift_card_total": 0,
    "gift_cards": Array [],
    "id": StringMatching /\\^cart_\\*/,
    "idempotency_key": null,
    "items": Array [
      Object {
        "allow_discounts": true,
        "cart_id": StringMatching /\\^cart_\\*/,
        "claim_order_id": null,
        "created_at": Any<String>,
        "description": "Small",
        "fulfilled_quantity": null,
        "has_shipping": false,
        "id": StringMatching /\\^item_\\*/,
        "is_giftcard": false,
        "metadata": Object {},
        "order_id": null,
        "quantity": 1,
        "returned_quantity": null,
        "shipped_quantity": null,
        "should_merge": true,
        "swap_id": null,
        "thumbnail": null,
        "title": "Basic Tee",
        "unit_price": 1000,
        "updated_at": Any<String>,
        "variant": Object {
          "allow_backorder": false,
          "barcode": null,
          "created_at": Any<String>,
          "deleted_at": null,
          "ean": null,
          "height": null,
          "hs_code": null,
          "id": StringMatching /\\^variant_\\*/,
          "inventory_quantity": 1,
          "length": null,
          "manage_inventory": true,
          "material": null,
          "metadata": null,
          "mid_code": null,
          "origin_country": null,
          "product_id": StringMatching /\\^prod_\\*/,
          "sku": null,
          "title": "Small",
          "upc": null,
          "updated_at": Any<String>,
          "weight": null,
          "width": null,
        },
        "variant_id": StringMatching /\\^variant_\\*/,
      },
    ],
    "metadata": null,
    "payment": null,
    "payment_authorized_at": null,
    "payment_id": null,
    "payment_sessions": Array [],
    "region": Object {
      "countries": Array [],
      "created_at": Any<String>,
      "currency_code": "usd",
      "deleted_at": null,
      "fulfillment_providers": Array [],
      "id": StringMatching /\\^reg_\\*/,
      "metadata": null,
      "name": "United States",
      "payment_providers": Array [],
      "tax_code": null,
      "tax_rate": "0",
      "updated_at": Any<String>,
    },
    "region_id": StringMatching /\\^reg_\\*/,
    "shipping_address": null,
    "shipping_address_id": null,
    "shipping_methods": Array [],
    "shipping_total": 0,
    "subtotal": 1000,
    "tax_total": 0,
    "total": 1000,
    "type": "default",
    "updated_at": Any<String>,
  },
}
`;

exports[`GetCustomersCustomer 1`] = `
Object {
  "customer": Object {
    "billing_address": null,
    "billing_address_id": null,
    "created_at": Any<String>,
    "deleted_at": null,
    "email": "ada@com.com",
    "first_name": "Ada",
    "has_account": true,
    "id": StringMatching /\\^cus_\\*/,
    "last_name": "Lovelace",
    "metadata": null,
    "phone": null,
    "shipping_addresses": Array [],
    "updated_at": Any<String>,
  },
}
`;

exports[`GetGiftCardsGiftCard 1`] = `
Object {
  "gift_card": Object {
    "balance": 500,
    "code": "1J3U-WXO0-SZ0H-O5QU",
    "id": StringMatching /\\^gift_\\*/,
    "region": Object {
      "created_at": Any<String>,
      "currency_code": "usd",
      "deleted_at": null,
      "fulfillment_providers": Array [],
      "id": StringMatching /\\^reg_\\*/,
      "metadata": null,
      "name": "United States",
      "payment_providers": Array [],
      "tax_code": null,
      "tax_rate": "0",
      "updated_at": Any<String>,
    },
    "value": 1000,
  },
}
`;

exports[`GetOrdersOrder 1`] = `
Object {
  "order": Object {
    "cart_id": null,
    "created_at": Any<String>,
    "currency_code": "dkk",
    "customer": ObjectContaining {
      "created_at": Any<String>,
      "id": StringMatching /\\^cus_\\*/,
      "updated_at": Any<String>,
    },
    "customer_id": StringMatching /\\^cus_\\*/,
    "discount_total": 0,
    "discounts": Array [],
    "display_id": 1,
    "email": "ada@com.com",
    "fulfillment_status": "not_fulfilled",
    "fulfillments": Array [],
    "gift_card_total": 0,
    "gift_card_transactions": Array [],
    "gift_cards": Array [],
    "id": StringMatching /\\^order_\\*/,
    "items": Array [],
    "payment_status": "not_paid",
    "payments": Array [],
    "refunded_total": 0,
    "refunds": Array [],
    "region": ObjectContaining {
      "created_at": Any<String>,
      "id": StringMatching /\\^reg_\\*/,
      "updated_at": Any<String>,
    },
    "region_id": StringMatching /\\^reg_\\*/,
    "shipping_address": null,
    "shipping_methods": Array [],
    "shipping_total": 0,
    "status": "pending",
    "subtotal": 0,
    "swaps": Array [],
    "tax_rate": 0,
    "tax_total": 0,
    "total": 0,
  },
}
`;

exports[`GetProductsProduct 1`] = `
Object {
  "product": Object {
    "collection": null,
    "collection_id": null,
    "created_at": Any<String>,
    "deleted_at": null,
    "description": null,
    "discountable": true,
    "handle": "basic-tee",
    "height": null,
    "hs_code": null,
    "id": StringMatching /\\^prod_\\*/,
    "images": Array [],
    "is_giftcard": false,
    "length": null,
    "material": null,
    "metadata": null,
    "mid_code": null,
    "options": Array [
      Object {
        "created_at": Any<String>,
        "deleted_at": null,
        "id": StringMatching /\\^opt_\\*/,
        "metadata": null,
        "product_id": StringMatching /\\^prod_\\*/,
        "title": "Size",
        "updated_at": Any<String>,
        "values": Array [
          Object {
            "created_at": Any<String>,
            "deleted_at": null,
            "id": StringMatching /\\^optval_\\*/,
            "metadata": null,
            "option_id": StringMatching /\\^opt_\\*/,
            "updated_at": Any<String>,
            "value": "Size",
            "variant_id": StringMatching /\\^variant_\\*/,
          },
        ],
      },
    ],
    "origin_country": null,
    "profile_id": StringMatching /\\^sp_\\*/,
    "status": "draft",
    "subtitle": null,
    "tags": Array [],
    "thumbnail": null,
    "title": "Basic Tee",
    "type": null,
    "type_id": null,
    "updated_at": Any<String>,
    "variants": Array [
      Object {
        "allow_backorder": false,
        "barcode": null,
        "created_at": Any<String>,
        "deleted_at": null,
        "ean": null,
        "height": null,
        "hs_code": null,
        "id": StringMatching /\\^variant_\\*/,
        "inventory_quantity": 1,
        "length": null,
        "manage_inventory": true,
        "material": null,
        "metadata": null,
        "mid_code": null,
        "origin_country": null,
        "prices": Array [
          Object {
            "amount": 8000,
            "created_at": Any<String>,
            "currency_code": "usd",
            "deleted_at": null,
            "id": StringMatching /\\^ma_\\*/,
            "region_id": null,
            "sale_amount": null,
            "updated_at": Any<String>,
            "variant_id": StringMatching /\\^variant_\\*/,
          },
        ],
        "product_id": StringMatching /\\^prod_\\*/,
        "sku": null,
        "title": "One Size",
        "upc": null,
        "updated_at": Any<String>,
        "weight": null,
        "width": null,
      },
    ],
    "weight": null,
    "width": null,
  },
}
`;

exports[`GetRegionsRegion 1`] = `
Object {
  "region": Object {
    "countries": Array [],
    "created_at": Any<String>,
    "currency_code": "dkk",
    "deleted_at": null,
    "fulfillment_providers": Array [
      Object {
        "id": "test-ful",
        "is_installed": true,
      },
    ],
    "id": StringMatching /\\^reg_\\*/,
    "metadata": null,
    "name": Any<String>,
    "payment_providers": Array [
      Object {
        "id": "test-pay",
        "is_installed": true,
      },
    ],
    "tax_code": null,
    "tax_rate": "0",
    "updated_at": Any<String>,
  },
}
`;

exports[`GetSwapsSwap 1`] = `
Object {
  "swap": Object {
    "allow_backorder": false,
    "canceled_at": null,
    "cart_id": StringMatching /\\^cart_\\*/,
    "confirmed_at": null,
    "created_at": Any<String>,
    "deleted_at": null,
    "difference_due": null,
    "fulfillment_status": "not_fulfilled",
    "id": StringMatching /\\^swap_\\*/,
    "idempotency_key": Any<String>,
    "metadata": null,
    "no_notification": true,
    "order_id": StringMatching /\\^order_\\*/,
    "payment_status": "not_paid",
    "shipping_address_id": null,
    "updated_at": Any<String>,
  },
}
`;
